protos/flyteidl/admin/node_execution.proto

```diff
@@ -79,6 +83,12 @@ message NodeExecution {
 
 // Represents additional attributes related to a Node Execution
 message NodeExecutionMetaData {
+    // Node executions are grouped depending on retries of the parent, and other scenarios
+    // Group id is unique within the context of a parent node.
+    string group_id = 1;
```

Comment:
I take this to imply that if I query for the children of a NodeExecution, I will get all children across all attempts and then I can sort them into buckets by this value on the client side?

Reply:
Exactly. @schottra  This is to prevent dummy nodes, and help with the visualization. We will return all executions and you can use the group id to group and show.